components/clm/src/data_types/ColumnDataType.F90

```diff
@@ -4516,19 +4517,29 @@ subroutine col_ps_restart ( this,  bounds, ncid, flag, cnstate_vars)
               errMsg(__FILE__, __LINE__))
           end if
 
-          do j = 1, nlevdecomp
-             rootfr(j) = exp(-3.0* zsoi(j))
-          end do
-          rootfr_tot = 0._r8
-          do j = 1, nlevdecomp
-             rootfr_tot = rootfr_tot + rootfr(j)
-          end do
-          do j = 1, nlevdecomp
-             pinit_prof(j) = rootfr(j) / rootfr_tot / dzsoi_decomp(j) ! 1/m
-          end do
-
           do c = bounds%begc, bounds%endc
              if (use_vertsoilc) then
+                ! calculate P initializtation profile
+                do j = 1, 6 ! top 50 cm
```

Comment:
Instead of assuming that layers 1-6 correspond to top 50 cm, add code outside the c-loop to figure out the j_index that corresponds to top 50cm based on `zsoi` or `zisoi`

Reply:
I added j_depth variable to store the index of 50cm depth soil layer